server/resourcemanager/encoding.go

```diff
@@ -0,0 +1,68 @@
+package resourcemanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+)
+
+type encoder interface {
+	Marshal(in interface{}) (out []byte, err error)
+	Unmarshal(in []byte, out interface{}) (err error)
+	Accepts(contentType string) bool
+	ResponseContentType() string
+}
+
+var encoders = []encoder{
```

Comment:
Registering encoders helps reduce coupling in the code by passing around specific instances instead of having `switch contentType`s everywhere